rust/datafusion/src/datasource/datasource.rs

```diff
@@ -48,9 +66,19 @@ pub trait TableProvider {
         &self,
         projection: &Option<Vec<usize>>,
         batch_size: usize,
+        filters: &[Expr],
```

Comment:
If we don't like passing around `&[]` all over the place, we could name `scan` --> `scan_with_filter` and add a default implementation of `scan` 

That would also have the nice benefit of keeping the signatures the same for existing TableProvider implementations. I do not feel strongly about this

Reply:
I think I'll leave this for the moment; on reflection, perhaps it'd be better to come up with a more future-proof design if we anticipate needing to expose more info to providers at scan-time; something along the lines of passing a "ScanArgs" struct rather than separate function args here?